include/zephyr/net/mqtt.h

```diff
@@ -133,6 +137,32 @@ enum mqtt_conn_return_code {
 	MQTT_NOT_AUTHORIZED                     = 0x05
 };
 
+/** @brief MQTT 5.0 CONNACK reason codes (MQTT 5.0, chapter 3.2.2.2). */
+enum mqtt_connect_reason_code {
```

Comment:
Some confusing naming here, `connect` vs `connack` vs `conn`. Not an issue, just an observation.

Reply:
That should be `connack` everywhere indeed, I've fixed the new code, however for the existing type - I would keep it as is not to break the API, unless we have a really strong push to do so.